### DISCUSSION TITLE
How does Dark Reader fix text aliasing for the FIlter modes?

### DISCUSSION THREAD

**Discussion by God-damnit-all** [darkreader/darkreader#8177] [Q&A]
With css, you can do `filter: invert(1) hue(180deg)` to invert a site's luminosity, but it makes the aliasing around the text look a bit weird. Dark Reader can similarly invert a website's luminosity with the filter mode, but somehow the text aliasing still looks crisp. How does it accomplish this, and can it even be done in plain CSS?

**Comment by alexanderby** [comment]
That's a magic. No. The browser disables sub-pixel rendering for CSS filters. But sometimes it fails, e.g. for PDFs.
